src/config.h.in

```diff
@@ -0,0 +1,4 @@
+#if XAPIAN_MAJOR_VERSION >= 2 || XAPIAN_MINOR_VERSION >=3 || (XAPIAN_MINOR_VERSION == 3 && XAPIAN_REVISION >= 6)
```

Comment:
Probably worth putting a quick comment at the top of this. In particular, I think you should note that this isn't automatically generated (some autotools projects have `config.h.in` generated automatically from `configure.ac`).

You actually don't need it to be `.in`, thinking about it. You aren't using any `@@…@@` replacements, so it'll probably just work as a normal header file. You probably also want `#include "xapian.h"` at the top, so they can be included in either order in source files.